incompatibility/incomp-scanning.go

```diff
@@ -16,56 +16,97 @@ package incompatibility
 
 import (
 	"errors"
+	"fmt"
+	"log"
 	"path/filepath"
 	"strings"
 
+	"github.com/googleapis/gnostic-grpc/search"
 	"github.com/googleapis/gnostic-grpc/utils"
 	openapiv3 "github.com/googleapis/gnostic/openapiv3"
 	plugins "github.com/googleapis/gnostic/plugins"
 	"google.golang.org/protobuf/proto"
-	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
 type Report int
 
 const (
 	BaseIncompatibility_Report = iota
-	ID_Report
+	FileDescriptive_Report
 )
 
-// Create a report of incompatibilities, write protobuf message to
-// environment, bool indicates if detailed incompatibility report is
-// desired.
-func CreateIncompReport(env *plugins.Environment, reportType Report) {
-
-	// Generate Base Incompatibility Report
-
-	// If indicated by report type associate incompatibilities with line
-	// references, etc. and generate an ID_REPORT
+// Runs incompatibility scanning under gnostic envirionment
+func GnosticIncompatibiltyScanning(env *plugins.Environment, reportType Report) {
 	for _, model := range env.Request.Models {
 		if model.TypeUrl != "openapi.v3.Document" {
 			continue
 		}
+
+		// Format into digestable object
 		openAPIdocument := &openapiv3.Document{}
 		err := proto.Unmarshal(model.Value, openAPIdocument)
 		env.RespondAndExitIfError(err)
-		incompatibilityReport := ScanIncompatibilities(openAPIdocument, env.Request.SourceName)
-		writeProtobufMessage(incompatibilityReport, env)
+
+		createdFile, reportErr := createIncompReport(openAPIdocument, env.Request.SourceName, reportType)
+		env.RespondAndExitIfError(reportErr)
+		env.Response.Files = append(env.Response.Files, createdFile)
+
 		env.RespondAndExit()
 	}
 	env.RespondAndExitIfError(errors.New("no supported models for incompatibility reporting"))
 }
 
-func writeProtobufMessage(m protoreflect.ProtoMessage, env *plugins.Environment) {
+// Creates and formats a specified incompatibility report under plugin.file object
+func createIncompReport(doc *openapiv3.Document, filePath string, reportType Report) (*plugins.File, error) {
+	//Generate Base Incompatibility Report
+	BaseReport := ScanIncompatibilities(doc, filePath)
```

Comment:
perhaps we can name this method baseReport to be inline with and reflect detailReport?

Reply:
named to createAndFormatReport since this function could format a base report or detail report